### PR TITLE
Add iter_elements helper function

### PR DESCRIPTION
**Pull request by daniel-thom**

### CODE REVIEW

**Comment by kdheepak** [comment]
Thanks for the PR! This will be a useful addition.

File: opendssdirect/utils.py [hunk 1]
```diff
@@ -279,3 +279,44 @@ def xycurves_to_dataframe(dss=None):
...
+        print(reg_control["name"])
+
+    >>> def get_reg_controls():
+        return list(iter_elements(dss.RegControls, get_reg_control))
+
+    # Create a list of RegControl dictionaries.
+    >>> reg_controls = get_reg_controls()
+        
+    """
+    element_class.First()
```
Comment (daniel-thom) [inline opendssdirect/utils.py]: Is there a way to validate whether this will work before we call it?  For example, you can't do this for Bus.

Comment (kdheepak) [reply opendssdirect/utils.py]: Not at the moment, but with the impending rewrite we should be able to do it.

**Comment by daniel-thom** [comment]
@kdheepak I went a different direction that I prefer. Let me know what you think.  If you also like it then we should squash the commits.

**Comment by kdheepak** [comment]
We already have something very similar. Maybe we should modify that instead?

https://github.com/dss-extensions/OpenDSSDirect.py/blob/1e64f25d047caf7f2f1013e51ff3b304f39cda1a/opendssdirect/utils.py#L16-L32

I think this PR is an improvement over what we currently have in the package.

**Comment by daniel-thom** [comment]
Sorry, I didn't notice the existing class - even though it's in the same file :(.

The two solutions are slightly different.  The existing solution provides iteration across elements over a single attribute.  The new one allows the user to do whatever they want with each element.  You could keep both of them.  You could also make the existing one use the new one.  Something like:

```
def Iterator(module, function):
    assert inspect.ismodule(module), "{module} must be of type module".format(
        module=module
    )
    for elem in ElementIterator(module):
        yield getattr(module, function)
```

**Comment by PMeira** [comment]
Hm... I don't mind (merging) this at all, but maybe it's time to work on #55 and do this more cleanly. It would be fine to merge this since the alternative is more invasive.

I'll try to work on #55 this week after DSS C-API 0.10.5 is out.

**Comment by daniel-thom** [comment]
@PMeira I don't think there is any rush to add this functionality.  I wrote this because I was getting tired of seeing people continually repeat calls to `.First()` and `.Next()`.  I prefer your example of

```
buses = DSS.ActiveCircuit.Buses
for bus in buses:
    print(bus.Name, bus.puVoltages)
```
to what  I have here with `to_namedtuples` or `ElementIterator`.

If you are going to suggest that users change how they are iterating over elements then I think it's better to wait for the final solution.

**Comment by daniel-thom** [comment]
Closing per discussion above.

**Pull request closed without merging**